src/components/HTMLEngineProvider/HTMLRenderers/PreRenderer/BasePreRenderer.js

```diff
@@ -0,0 +1,29 @@
+import React, {forwardRef} from 'react';
+import {ScrollView} from 'react-native-gesture-handler';
+import {View} from 'react-native';
+import _ from 'underscore';
+import htmlRendererPropTypes from '../htmlRendererPropTypes';
+
+class BasePreRenderer extends React.Component {
+    render() {
+        const TDefaultRenderer = this.props.TDefaultRenderer;
+        const defaultRendererProps = _.omit(this.props, ['TDefaultRenderer']);
+        return (
+            <ScrollView ref={this.props.innerRef} horizontal>
+                <View onStartShouldSetResponder={() => true}>
+                    <TDefaultRenderer
+                     // eslint-disable-next-line react/jsx-props-no-spreading
+                        {...defaultRendererProps}
+                    />
+                </View>
+            </ScrollView>
+        );
+    }
+}
```

Comment:
Looks like a perfect candidate for the functional components. Please read the style guide and correct all style issues.

Reply:
> Looks like a perfect candidate for the functional components. Please read the style guide and correct all style issues.

@parasharrajat i refactored. Please re-check